src/data/finsupp/well_founded.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2022 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import data.dfinsupp.well_founded
+import data.finsupp.lex
+
+/-!
+# Well-foundedness of the lexicographic and product orders on `finsupp`
+
+-/
+
+variables {α N : Type*}
+
+namespace finsupp
+
+variables [hz : has_zero N] {r : α → α → Prop} {s : N → N → Prop}
+  (hbot : ∀ ⦃n⦄, ¬ s n 0) (hs : well_founded s)
+include hbot hs
+
+lemma lex.acc (x : α →₀ N) (h : ∀ a ∈ x.support, acc (rᶜ ⊓ (≠)) a) : acc (finsupp.lex r s) x :=
```

Comment:
Same as for `dfinsupp`: explain the use of the complementary relation.

Reply:
Added a reference to the module docstring of *dfinsupp/well_founded*.